docs/topics/whatsnew1620.md

```diff
@@ -0,0 +1,851 @@
+The syntactic form comes from a notation of [intersection types](https://en.wikipedia.org/wiki/Intersection_type) and is now limited to a type parameter with nullable upper bounds on the left side of `&` and non-nullable `Any` on the right side:
+
+```kotlin
+fun <T> elvisLike(x: T, y: T & Any): T & Any = x ?: y
+
+fun main() {
+    // OK
+    elvisLike<String>("", "").length
+    // Error: 'null' cannot be a value of a non-null type
+    elvisLike<String>("", null).length
+
+    // OK
+    elvisLike<String?>(null, "").length
+    // Error: 'null' cannot be a value of a non-null type
+    elvisLike<String?>(null, null).length
+}
+```
+{validate="false"}
+
+Set the language version to `1.7` to enable the feature:
+
+<tabs group="build-script">
+<tab title="Kotlin" group-key="kotlin">
+
+```kotlin
+kotlin {
+    sourceSets.all {
+        languageSettings.apply {
+            languageVersion = "1.7"
+        }
+    }
+}
+```
+
+</tab>
+<tab title="Groovy" group-key="groovy">
+
+```groovy
+kotlin {
+    sourceSets.all {
+        languageSettings {
+            languageVersion = '1.7'
+        }
+    }
+}
+```
+
+</tab>
+</tabs>
+
+Learn more about definitely non-nullable types in [the KEEP](https://github.com/Kotlin/KEEP/blob/c72601cf35c1e95a541bb4b230edb474a6d1d1a8/proposals/definitely-non-nullable-types.md).
+
+## Kotlin/JVM
+
+In Kotlin 1.6.20 introduces:
+
+* Compatibility improvements of default methods in JVM interfaces: [new `@JvmDefaultWithCompatibility` annotation for interfaces](#new-jvmdefaultwithcompatibility-annotation-for-interfaces) and [compatibility changes in the `-Xjvm-default` modes](#compatibility-changes-in-the-xjvm-default-modes)
+* [Support for parallel compilation of a single module in the JVM backend](#support-for-parallel-compilation-of-a-single-module-in-the-jvm-backend)
+* [Support for callable references to function interface constructors](#support-for-callable-references-to-function-interface-constructors)
+
+### New @JvmDefaultWithCompatibility annotation for interfaces
+
+Kotlin 1.6.20 introduces the [new annotation `@JvmDefaultWithCompatibility`](https://github.com/JetBrains/kotlin/blob/master/libraries/stdlib/jvm/src/kotlin/jvm/JvmDefault.kt#L53): use it along with the `-Xjvm-default=all` compiler option [to create the default method in JVM interface](java-to-kotlin-interop.md#default-methods-in-interfaces) for any non-abstract member in any Kotlin interface.
+
+If there are clients that use your Kotlin interfaces compiled without the `-Xjvm-default=all` option, they may be binary-incompatible with the code compiled with this option.
+Before Kotlin 1.6.20, to avoid this compatibility issue,  [the recommended approach](https://blog.jetbrains.com/kotlin/2020/07/kotlin-1-4-m3-generating-default-methods-in-interfaces/#JvmDefaultWithoutCompatibility) was to use the `-Xjvm-default=all-compatibility` mode and also the `@JvmDefaultWithoutCompatibility` annotation for interfaces that didn't need this type of compatibility.
```

Comment:
```suggestion
Before Kotlin 1.6.20, to avoid this compatibility issue, the [recommended approach](https://blog.jetbrains.com/kotlin/2020/07/kotlin-1-4-m3-generating-default-methods-in-interfaces/#JvmDefaultWithoutCompatibility) was to use the `-Xjvm-default=all-compatibility` mode and also the `@JvmDefaultWithoutCompatibility` annotation for interfaces that didn't need this type of compatibility.
```